docs/guides/javascript.rst

```diff
@@ -302,7 +302,22 @@ Emit a hook event in the event system.
 .. code:: js
 
     // allow other plugins to alter value
-    value = elgg.trigger_hook('my_plugin:filter', 'value', {}, value);
+    value = elgg.trigger_hook('filter', 'my_plugin:value', {}, value);
+
+
+``elgg.filter()``
+
+Pass a value through the "filter" hook.
+
+.. code:: js
+
+    // allow other plugins to alter value
+    value = elgg.filter('my_plugin:value', value);
+
+    // define a class method that plugins can alter
+    Foo.prototype.init = elgg.filter('Foo::init', function () {
```

Comment:
Yikes :/ This feels a little too fancy. In particular plugin hook configuration is local/runtime info while I usually associate class definitions with global/static/compile time. I realize these can be conflated/mixed/matched in JS, but the dependency graph here feels backwards.